espnet/nets/pytorch_backend/frontends/beamformer.py

```diff
@@ -0,0 +1,158 @@
+import torch
+from torch_complex import functional as FC
+from torch_complex.tensor import ComplexTensor
+
+
+# Ported from https://github.com/fgnt/nn-gev
+
+
+def get_power_spectral_density_matrix(xs: ComplexTensor, mask: torch.Tensor,
+                                      normalization=True) -> ComplexTensor:
+    """Return cross-channel power spectral density (PSD) matrix
+        (a,k.a spatial covariance matrices)
+
+    Args:
+        xs (ComplexTensor): (..., F, C, T)
+        mask (torch.Tensor): (..., F, C, T)
+        normalization (bool):
+    Returns
+        psd (ComplexTensor): (..., F, C, C)
+
+    """
+    # outer product: (..., C_1, T) x (..., C_2, T) -> (..., T, C, C_2)
+    psd_Y = FC.einsum('...ct,...et->...tce', [xs, xs.conj()])
+
+    # Averaging mask along C: (..., C, T) -> (..., T)
+    mask = mask.mean(dim=-2)
+
+    # Normalized mask along T: (..., T)
+    if normalization:
+        mask = mask / mask.sum(dim=-1)[..., None]
```

Comment:
Does this consider the padding well?

Reply:
If assuming  the tensor is padded with zero, the summation along the time axis is same regardless of the padding length. 
This might be a little confusing, so I'll put a comment there.